src/tls13.c

```diff
@@ -4265,23 +4265,35 @@ static int CheckPreSharedKeys(WOLFSSL* ssl, const byte* input, word32 helloSz,
         else
     #endif
         {
-            if ((modes & (1 << PSK_KE)) == 0)
+            if ((modes & (1 << PSK_KE)) == 0) {
+                WOLFSSL_MSG("psk_ke mode does not allow key share");
                 return PSK_KEY_ERROR;
+            }
             ssl->options.noPskDheKe = 1;
             ssl->arrays->preMasterSz = 0;
 
             *usingPSK = 1;
         }
     }
-#ifdef WOLFSSL_PSK_ID_PROTECTION
     else {
-    #ifndef NO_CERTS
+        /* RFC8446: 4.2.9. Pre-Shared Key Exchange Modes:
```

Comment:
This else condition is when we are not using PSK.
We check that we have certificates to ensure that we can complete handshake.
I don't see a need for this code.